### DISCUSSION TITLE
Refresh interval but only when idle

### DISCUSSION THREAD

**Discussion by hazae41** [vercel/swr#1558] [Ideas]
I would like a page that displays some feed, and refresh it automatically. However, it should not refresh while the user is active on the page, because it would be disturbing.

Is there a way to have a refresh interval, but only if the user is NOT doing anything e.g. scrolling the page, dragging mouse, typing something, etc.?

I could use the new Idle Detection API or listen for mouse and scroll events, but isn't there already a such feature in SWR?

**Comment by shuding** [comment]
In short: SWR doesn't have the ability now, but this is actually a good suggestion! There're two related options:
- `refreshWhenHidden`
- `refreshWhenOffline`

So it's great to have `refreshWhenIdle` too, and even more you should be able to provide an idle event listener (to customize).

~Going to convert this into an issue 👍~

Created #1562.
